meson.build

```diff
@@ -373,12 +377,21 @@ else
     libass_for_tools = libass
 endif
 
-if get_option('test')
+if get_option('test').enabled()
     subdir('test')
 endif
-if get_option('profile')
+if get_option('compare').require(default_library != 'shared').enabled()
```

Comment:
Why this is disabled in shared build?

Reply:
Because it _requires_ access to libass-internal symbols which is not possible with a proper dynamic library.
The autotools build has a way to deal with this which can eventually be adapted into the meson setup, but for now just getting tests at all and restricting this is fine imo

Reply:
If a tool or test depends on internal symbols, indeed it shouldn't like with libass directly. Meson has easy way to handle this case, without splitting into static libraries, `libass.extract_all_objects(recursive: true)`  or `libass.extract_objects(...)` for more fine-grained objects to link.